app/stores/DocumentsStore.ts

```diff
@@ -660,7 +660,11 @@ export default class DocumentsStore extends BaseStore<Document> {
 
   @action
   removeCollectionDocuments(collectionId: string) {
-    const documents = this.inCollection(collectionId);
+    // drafts are to be detached from collection rather than deleted, hence excluded here
+    const documents = filter(
+      this.inCollection(collectionId),
+      (d) => !!d.publishedAt
+    );
```

Comment:
This retains drafts in two other cases, `entities` and `collections.remove_user` of `WebsocketProvider`. Hope that's fine :crossed_fingers: